### DISCUSSION TITLE
Declarative schema for databases

### DISCUSSION THREAD

**Discussion by MichelJonkman** [laravel/framework#42489] [Ideas] [answered]
I could not find an existing discussion regarding this topic so I thought I'd start one.

Since Magento 2.3 they have a feature called "[Declarative schema](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/declarative-schema/)". This is basically just an XML file defining what the table should look like, eliminating the need for migration files. The implementation in Laravel could be similar and use XML as well, or is could use a similar style to how migrations are right now.

Let me know what you think! Any issues that would come up that I haven't thought about?

**Comment by dkarvounaris** [comment, marked as answer]
There is https://github.com/laravel/framework/discussions/39855 and https://github.com/laravel/ideas/issues/944 in that regard.

Maybe the discussion about it could be revived?
In a recent project, migrations got out of control and such a feature would simplify everything and would help with complexity.

**Reply by MichelJonkman** [reply]
At the moment I'm planning on creating my own approach, maybe when that is done I can propose it again!

**Comment by IgorsKolcins** [comment]
It's 2025 now. Does anybody know if there's a solution to this?

**Reply by MichelJonkman** [reply]
This exists but there's no official Laravel integration (yet)

**Comment by shaedrich** [comment]
This could also be an alternative output format for dumping the schema
